xbmc/platform/android/utils/AndroidTVPerformance.cpp

```diff
@@ -0,0 +1,771 @@
+  // For 4K capable devices, enable 4K GUI
+  if (m_has4K && m_gpuLevel >= 5)
+  {
+    settings->SetInt("videoscreen.guiresolution", 0);  // Auto resolution
+  }
+  
+  // Memory settings based on device capabilities
+  if (m_totalMemory < 1LL * 1024 * 1024 * 1024)  // Less than 1GB RAM
+  {
+    // Low memory device
+    CLog::Log(LOGINFO, "AndroidTVPerformance: Configuring for low-memory device");
+    settings->SetInt("network.cachemembuffersize", 5242880);  // 5MB network cache
+  }
+  else if (m_totalMemory < 2LL * 1024 * 1024 * 1024)  // 1-2GB RAM
+  {
+    // Medium memory device
+    CLog::Log(LOGINFO, "AndroidTVPerformance: Configuring for medium-memory device");
+    settings->SetInt("network.cachemembuffersize", 20971520);  // 20MB network cache
+  }
+  else  // 2GB+ RAM
+  {
+    // High memory device
+    CLog::Log(LOGINFO, "AndroidTVPerformance: Configuring for high-memory device");
+    settings->SetInt("network.cachemembuffersize", 41943040);  // 40MB network cache
+  }
+}
+
+void CAndroidTVPerformance::MonitorPerformance()
+{
+  // Check if video is playing and optimize if needed
+  if (g_application.GetAppPlayer().IsPlaying() && !m_isVideoPlaying && 
+      g_application.GetAppPlayer().IsPlayingVideo())
+  {
+    // Video just started
+    int width = 0;
+    int height = 0;
+    float fps = 0.0f;
+    bool isHDR = false;
+    
+    // Get video details
+    g_application.GetAppPlayer().GetVideoResolution(width, height);
+    fps = g_application.GetAppPlayer().GetVideoFps();
+    
+    // Check if HDR
+    CVideoSettings videoSettings = g_application.GetAppPlayer().GetVideoSettings();
+    isHDR = videoSettings.m_HDR;
+    
+    // Optimize for this video
+    OptimizeVideoPlayback(width, height, fps, isHDR);
+  }
+  else if (!g_application.GetAppPlayer().IsPlayingVideo() && m_isVideoPlaying)
+  {
+    // Video playback ended
+    RestoreAfterVideoPlayback();
+  }
+  
+  // Periodically manage memory
+  if (m_performanceMonitor.IsRunning() && m_performanceMonitor.GetElapsedSeconds() > 300)  // Every 5 minutes
+  {
+    ManageMemory(false);
+    m_performanceMonitor.Reset();
+  }
+  
+  // Restart timer for next check
+  m_monitorTimer.Start();
+}
+
+bool CAndroidTVPerformance::HasUHDCapability()
+{
+  return m_has4K;
+}
+
+bool CAndroidTVPerformance::HasHDRCapability()
+{
+  return m_hasHDR;
+}
+
+void CAndroidTVPerformance::ConfigureDRMPlayback()
+{
+  if (CServiceBroker::GetSettingsComponent())
+  {
+    auto settings = CServiceBroker::GetSettingsComponent()->GetSettings();
+    
+    // Configure DRM settings for secure playback
+    settings->SetBool("videoplayer.useprimedecoder", true);
+    
+    // Check if we need Widevine DRM
+    bool hasWidevine = false;
+    JNIEnv* env = xbmc_jnienv();
+    if (env)
+    {
+      jobject activity = CXBMCApp::Get().getActivity();
+      jclass activityClass = env->GetObjectClass(activity);
+      jmethodID hasWidevineDrm = env->GetMethodID(activityClass, "hasWidevineDrm", "()Z");
```

Comment:
here it looks like it only controls for widevine, but for all other supported DRM?
you can currently play Widevine, PlayReady, Wiseplay, Clearkey